otel-extension/src/main/java/akka/actor/impl/ActorCellReceivedAdvice.java

```diff
@@ -0,0 +1,44 @@
+package akka.actor.impl;
+
+import akka.actor.ActorContext;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.field.VirtualField;
+import io.scalac.mesmer.core.util.Interval$;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.Instruments;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.InstrumentsProvider;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.impl.otel.ActorCellInstrumentationState;
+import java.util.Objects;
+import net.bytebuddy.asm.Advice;
+
+public class ActorCellReceivedAdvice {
+
+  @Advice.OnMethodEnter
+  public static long enter(@Advice.This ActorContext self) {
+
+    return System.nanoTime();
+  }
+
+  @Advice.OnMethodExit(onThrowable = Throwable.class)
+  public static void exit(
+      @Advice.This ActorContext self,
+      @Advice.Thrown Throwable exception,
+      @Advice.Enter long started) {
+    long interval = Interval$.MODULE$.toMillis(System.nanoTime() - started);
+
+    Attributes attrs = VirtualField.find(ActorContext.class, Attributes.class).get(self);
+    Instruments instruments = InstrumentsProvider.instance();
+    ActorCellInstrumentationState state =
+        VirtualField.find(ActorContext.class, ActorCellInstrumentationState.class).get(self);
+
+    if (Objects.nonNull(attrs) && Objects.nonNull(state)) {
+      instruments.processingTime().record(interval, attrs);
+      /*
+         Here we check it there was an exception and if TypedInstrumentation already taken care of this
+      */
```

Comment:
Theres
```suggestion

```

It's clear what the code is doing so the comment is redundant

Reply:
I disagree - I must say that this part took me a while to figure out from the old code that's why I left this comment and there is more in depth explanation of the process in `akka.actor.impl.typed.SupervisorHandleReceiveExceptionAdvice`  . We can change this to something else though like: 
```
       To keep failed messages metric consistent we must not increase the counter if type instrumentation already handled it
         Check akka.actor.impl.typed.SupervisorHandleReceiveExceptionAdvice for more details
```
WDYT @lgajowy

Reply:
Ok - now i see the intent more clearly and like this version better. 

Could you then extract this to a private method and utilize comment hyperlinks? IDEs will support that (and avoid name staying old when renaming). Something like this: 

```
/**
   * To keep failed messages metric consistent we must not increase the counter if type
   * instrumentation already handled it.
   *
   * @see akka.actor.impl.typed.SupervisorHandleReceiveExceptionAdvice for more details
   */
  private static void incrementFailedMessages(
      Throwable exception,
      Attributes attributes,
      Instruments instruments,
      ActorCellInstrumentationState state) {
    if (Objects.nonNull(exception) && !state.getAndResetFailed()) {
      instruments.failedMessages().add(1L, attributes);
    }
  }
```